src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -1972,7 +1972,12 @@ private void getFieldInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_MET
                     MethodDesc contextMethod = methodFromContext(pResolvedToken.tokenContext);
                     if (contextMethod == MethodBeingCompiled)
                     {
-                        FieldDesc runtimeDeterminedField = (FieldDesc)GetRuntimeDeterminedObjectForToken(ref pResolvedToken);
```

Comment:
Do you think this switch-out of the runtime determined field with the original field unique to this invocation of `GetRuntimeDeterminedObjectForToken`? Or should we encapsulate the behavior for R2R inside `GetRuntimeDeterminedObjectForToken` so other callers don't have to switch it out?

Reply:
For example, the return from `GetRuntimeDeterminedObjectForToken` can be treated as a FieldDesc here:
https://github.com/dotnet/corert/blob/57ad558d17d0338691a2e68c2055fca779432650/src/JitInterface/src/CorInfoImpl.cs#L2531

Reply:
Sadly, there are a couple of other invocations of GetRuntimeDeterminedObjectForToken and I verified before that not all can be short-circuited like this. I suspect that in some cases we'll need to carry the runtime determined parameters around and subsequently use them in some next level of signature emission or another lookup.

Reply:
I actually spent most of the last two days figuring out what is the right place to call GetRuntimeDeterminedObjectForToken in CPAOT compilation. I believe that, as of 4th commit, I have weeded out all (both) cases where a runtime determined object ended up leaking into the dependency graph subsequently causing a "NecessaryTypeNode" assertion failure (fields and delegate ctors). With this change I received the best Pri#0 run so far with just 111 failures.